README.md

```diff
@@ -20,53 +31,166 @@ This crate works with Cargo and can be found on
 bson = "0.14"
 ```
 
+## Overview of BSON Format
+
+BSON, short for Binary JSON, is a binary-encoded serialization of JSON-like documents.
+Like JSON, BSON supports the embedding of documents and arrays within other documents
+and arrays. BSON also contains extensions that allow representation of data types that
+are not part of the JSON spec. For example, BSON has a datetime type and a binary data type.
+
+```text
+// JSON equivalent
+{"hello": "world"}
+
+// BSON encoding
+\x16\x00\x00\x00                   // total document size
+\x02                               // 0x02 = type String
+hello\x00                          // field name
+\x06\x00\x00\x00world\x00          // field value
+\x00                               // 0x00 = type EOO ('end of object')
+```
+
+BSON is the primary data representation for [MongoDB](https://www.mongodb.com/), and this crate is used in the
+[`mongodb`](https://docs.rs/mongodb/0.10.0/mongodb/) driver crate in its API and implementation.
+
+For more information about BSON itself, see [bsonspec.org](http://bsonspec.org).
+
 ## Usage
 
-Prepare your struct for Serde serialization:
+### BSON values
+
+Many different types can be represented as a BSON value, including 32-bit and 64-bit signed
+integers, 64 bit floating point numbers, strings, datetimes, embedded documents, and more. To
+see a full list of possible BSON values, see the [BSON specification](http://bsonspec.org/spec.html). The various
+possible BSON values are modeled in this crate by the [`Bson`](https://docs.rs/bson/latest/bson/enum.Bson.html) enum.
+
+#### Creating [`Bson`](https://docs.rs/bson/latest/bson/enum.Bson.html) instances
+
+[`Bson`](https://docs.rs/bson/latest/bson/enum.Bson.html) values can be instantiated directly or via the
+[`bson!`](https://docs.rs/bson/latest/bson/macro.bson.html) macro:
 
 ```rust
-#[derive(Serialize, Deserialize, Debug)]
-pub struct Person {
-    #[serde(rename = "_id")]  // Use MongoDB's special primary key field name when serializing 
-    pub id: bson::oid::ObjectId,
-    pub name: String,
-    pub age: i32
-}
+let string = Bson::String("hello world".to_string());
+let int = Bson::I32(5);
+let array = Bson::Array(vec![Bson::I32(5), Bson::Boolean(false)]);
+
+let string: Bson = "hello world".into();
+let int: Bson = (5 as i32).into();
```

Comment:
Slight nit: instead of casting, you can do `5i32` to explicitly create the value as an i32. This also lets you omit the parentheses

Reply:
done